File: RxTools-library/src/main/java/com/vondear/rxtools/interfaces/onRepeatClickListener.java
package com.vondear.rxtools.interfaces;

import android.view.View;

import com.vondear.rxtools.RxUtils;
import com.vondear.rxtools.view.RxToast;

/**
 * Created by Vondear on 2017/7/24.
 * 重复点击的监听器
 */

public abstract class onRepeatClickListener implements View.OnClickListener {

    // 两次点击按钮之间的点击间隔不能少于1000毫秒
    private static final int MIN_CLICK_DELAY_TIME = 1000;

    public abstract void onRepeatClick(View v);

    @Override
    public void onClick(View v) {
        if (!RxUtils.isFastClick(MIN_CLICK_DELAY_TIME)) {
            onRepeatClick(v);
        }else{
            RxToast.normal("请不要重复点击");
            return;
        }
    }

}

File: RxKit/src/main/java/com/tamsiree/rxkit/interfaces/OnRepeatClickListener.kt
package com.tamsiree.rxkit.interfaces

import android.view.View
import com.tamsiree.rxkit.RxTool
import com.tamsiree.rxkit.view.RxToast

/**
 *
 * @author Tamsiree
 * @date 2017/7/24
 * 重复点击的监听器
 */
abstract class OnRepeatClickListener : View.OnClickListener {
    // 两次点击按钮之间的点击间隔不能少于1000毫秒
    private val MIN_CLICK_DELAY_TIME = 1000
    abstract fun onRepeatClick(v: View?)
    override fun onClick(v: View) {
        if (!RxTool.isFastClick(MIN_CLICK_DELAY_TIME)) {
            onRepeatClick(v)
        } else {
            RxToast.normal("请不要重复点击")
            return
        }
    }
}